enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -28,6 +30,100 @@ const (
 
 var shuttingDownLogOnce sync.Once
 
+type groupPriorityQueue struct {
+	*priority_queue.PriorityQueue
+	groupID string
+}
+
+type taskQueue struct {
+	// List of *groupPriorityQueue items.
+	pqs *list.List
+	// Map to allow quick lookup of a specific *groupPriorityQueue element in the pqs list.
+	pqByGroupID map[string]*list.Element
+	// The *groupPriorityQueue element from which the next task will be obtained.
+	// Will be nil when there are no tasks remaining.
+	currentPQ *list.Element
+	// Number of tasks across all queues.
+	numTasks int
+}
+
+func newTaskQueue() *taskQueue {
+	return &taskQueue{
+		pqs:         list.New(),
+		pqByGroupID: make(map[string]*list.Element),
+		currentPQ:   nil,
+	}
+}
+
+func (p *taskQueue) Enqueue(req *scpb.EnqueueTaskReservationRequest) {
```

Comment:
nit: maybe name the instance var t instead of p so it's less mental work to distinguish local lookups p.* from pq.* and pqEl.*?

Reply:
Done.